### PR TITLE
Support AWS Web Identity Provider

### PR DESCRIPTION
**Pull request by zjj2wry**
**How to categorize this PR?**

/area TODO
/kind TODO
/priority normal
/platform aws

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: pkg/aws/secret.go [hunk 1]
```diff
@@ -60,9 +60,7 @@ func ReadCredentialsSecret(secret *corev1.Secret) (*Credentials, error) {
 // NewClientFromSecretRef creates a new Client for the given AWS credentials from given k8s <secretRef> and
 // the AWS region <region>.
 func NewClientFromSecretRef(ctx context.Context, client client.Client, secretRef corev1.SecretReference, region string) (awsclient.Interface, error) {
-	credentials, err := GetCredentialsFromSecretRef(ctx, client, secretRef)
-	if err != nil {
-		return nil, err
-	}
+	credentials, _ := GetCredentialsFromSecretRef(ctx, client, secretRef)
```
Comment (tennix) [inline pkg/aws/secret.go:63]: Please add a comment here if ignoring the error.

File: pkg/controller/infrastructure/actuator_delete.go [hunk 2]
```diff
@@ -65,14 +65,19 @@ func Delete(
 		return fmt.Errorf("error while checking whether terraform config exists: %+v", err)
 	}
 
-	credentials, err := aws.GetCredentialsFromSecretRef(ctx, c, infrastructure.Spec.SecretRef)
-	if err != nil {
-		return err
-	}
+	credentials, _ := aws.GetCredentialsFromSecretRef(ctx, c, infrastructure.Spec.SecretRef)
```
Comment (tennix) [inline pkg/controller/infrastructure/actuator_delete.go:68]: ditto

File: pkg/controller/infrastructure/actuator_reconcile.go [hunk 3]
```diff
@@ -64,10 +64,7 @@ func Reconcile(
 	error,
 ) {
 
-	credentials, err := aws.GetCredentialsFromSecretRef(ctx, c, infrastructure.Spec.SecretRef)
-	if err != nil {
-		return nil, nil, err
-	}
+	credentials, _ := aws.GetCredentialsFromSecretRef(ctx, c, infrastructure.Spec.SecretRef)
```
Comment (tennix) [inline pkg/controller/infrastructure/actuator_reconcile.go:67]: ditto

File: pkg/controller/infrastructure/actuator_delete.go [hunk 4]
```diff
@@ -65,14 +65,19 @@ func Delete(
...
+	var awsClient awsclient.Interface
 
-	awsClient, err := awsclient.NewClient(string(credentials.AccessKeyID), string(credentials.SecretAccessKey), infrastructure.Spec.Region)
-	if err != nil {
-		return err
+	if credentials != nil {
```
Comment (tennix) [inline pkg/controller/infrastructure/actuator_delete.go:71]: Can we assume that when the above error is nil, the credentials would be non-nil, so we don't have to branch here with the `if-else` block?

Comment (zjj2wry) [reply pkg/controller/infrastructure/actuator_delete.go]: @tennix I update the code to ensure credentials is not null, then we can remove the `if-else` block

**Comment by zjj2wry** [comment]
@tennix @aylei PTAL

**Review by tennix — APPROVED** [review]
LGTM

**Review by aylei — APPROVED** [review]
lgtm

**Pull request merged**